### PR TITLE
chore: update contributing with more information on testing

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Description of changes

Updates contributing with more detailed instructions for running unit tests.

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -190,18 +192,54 @@ The "pre-commit" hook runs the [verify-commit](https://github.com/aws-amplify/am
 
 #### "pre-push" hook:
 
-The "pre-push" hook will build test files and run the  `split-e2e-tests` script to ensure the correct configuration file is generated for our CICD workflow.
+The "pre-push" hook will build test files and run the `split-e2e-tests` script to ensure the correct configuration file is generated for our CICD workflow.
```
Comment (sobolk) [inline CONTRIBUTING.md:195]:
```suggestion
The "pre-push" hook will build test files and run the `split-e2e-tests` script to ensure the correct configuration file is generated for our CI/CD workflow.
```

**Review by jhockett — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -190,18 +192,54 @@ The "pre-commit" hook runs the [verify-commit](https://github.com/aws-amplify/am
...
 To run the test suite:
 
 ```bash
 yarn test
 ```
 
+Unit tests can be run on a per-project basis:
+
+```bash
+yarn nx test amplify-category-function
```
Comment (sobolk) [inline CONTRIBUTING.md:210]: I wish we could `yarn test amplify-category-function` or something like this. I.e. don't `yarn` and `nx` at the same time.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -190,18 +192,54 @@ The "pre-commit" hook runs the [verify-commit](https://github.com/aws-amplify/am
...
+```
+
+Or, optionally,
```
Comment (sobolk) [inline CONTRIBUTING.md:213]:
```suggestion
or
```

**Review by sobolk — DISMISSED** [review]

**Review by jhockett — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (sobolk) [reply CONTRIBUTING.md:195]: This came back. Force pushes are evil.

**Review by sobolk — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**